File: expressions/finantial_calc.py
#Maisha Paredes, Finantial Calculator python

# print statement that welcomes user and tells us what program does

print("Welcome! This program will calculate your savings and how much you have left to spend. You can only type the number, don't put dollar signs!")

#ask user what their monthly income is (variable, an input)

income = float(input("What is your income?\n"))

#ask user what thier rent is (variable, an input)

rent = float(input("How much is your rent?\n"))
#ask user what their utilities is (variable, an input)

utilities = float(input("How much are your utilities?\n"))

#ask user what their groceries is (variable, an input)

groceries = float(input("How much are your groceries?\n"))

#ask user what their transportation is (variable, an input)

transport = float(input("How much is your transportation?\n"))

#calculate savings as 10% of income (income*.1) (variable)

savings = income * 0.1

#calculate spending as income-savings-rent-utilities-groceries-transport (variable)

spendings = income-savings-rent-utilities-groceries-transport
#calculate percent income of rent (rent/income *100)  (variable)
 rent_percentage = (rent/income) *100

#calculate percent income of utilities (utilities/income *100)  (variable)

utilities_percentage = (utilities/income) *100

#calculate percent income of groceries (groceries/income *100)  (variable)

groceries_percentage = (groceries / income) * 100

#calculate percent income of transportation (transportation/income *100)  (variable)
transport_percentage = (transport / income) * 100

#calculate percent income of spending (spending/income *100)  (variable)
spending_percentage = (spendings / income) * 100

#your rent is $XX.XX which is XX% of your income. (Print)
print(f"Your rent is ${rent:.2f} which is {rent_percentage:.2f}% of your income.")

#your utilities is $XX.XX which is XX% of your income. (Print)
print(f"Your utilities are ${utilities:.2f} which is {utilities_percentage:.2f}% of your income.")

#your groceries is $XX.XX which is XX% of your income. (Print)
print(f"Your groceries are ${groceries:.2f} which is {groceries_percentage:.2f}% of your income.")

#your transportation is $XX.XX which is XX% of your income. (Print)
print(f"Your transportation is ${transport:.2f} which is {transport_percentage:.2f}% of your income.")

#your savings are $XX.XX which is XX% of your income. (Print)
savings_percentage = (savings / income) * 100  # Calculate savings percentage
print(f"Your savings are ${savings:.2f} which is {savings_percentage:.2f}% of your income.")

#your spendings are $XX.XX which is XX% of your income. (Print)
print(f"Your spendings are ${spendings:.2f} which is {spending_percentage:.2f}% of your income.")

File: functions^^/finantialcalc.c
// Maisha Paredes, finantial calc c update thing
#include <stdio.h>  
  
float get_user_inputs() {  
    float income, rent, utilities, groceries, transport;  
  
    printf("What is your income?\n");  
    scanf("%f", &income);  
  
    printf("How much is your rent?\n");  
    scanf("%f", &rent);  
  
    printf("How much are your utilities?\n");  
    scanf("%f", &utilities);  
  
    printf("How much are your groceries?\n");  
    scanf("%f", &groceries);  
  
    printf("How much is your transportation?\n");  
    scanf("%f", &transport);  
    
    return income; 
}  
  
void calculate_percentages(float income, float rent, float utilities, float groceries, float transport, float *savings, float *spendings, float *rent_percentage, float *utilities_percentage, float *groceries_percentage, float *transport_percentage, float *spending_percentage) {  
    *savings = income * 0.1;    
    *spendings = income - *savings - rent - utilities - groceries - transport;  
    *rent_percentage = (rent / income) * 100;    
    *utilities_percentage = (utilities / income) * 100;   
    *groceries_percentage = (groceries / income) * 100;    
    *transport_percentage = (transport / income) * 100;    
    *spending_percentage = (*spendings / income) * 100;    
}  
  
void print_results(float savings, float rent, float rent_percentage, float utilities, float utilities_percentage, float groceries, float groceries_percentage, float transport, float transport_percentage, float spendings, float spending_percentage) {  
   // print statement that welcomes user and tells us what program does
    printf("Welcome to my calculator that will help you manage your monthly finances!\n");  
    printf("Your rent is \$%.2f which is %.2f%% of your income.\n", rent, rent_percentage);  
    printf("Your utilities are \$%.2f which is %.2f%% of your income.\n", utilities, utilities_percentage);  
    printf("Your groceries are \$%.2f which is %.2f%% of your income.\n", groceries, groceries_percentage);  
    printf("Your transportation is \$%.2f which is %.2f%% of your income.\n", transport, transport_percentage);  
    printf("Your savings are \$%.2f which is %.2f%% of your income.\n", savings, (savings / (rent + utilities + groceries + transport)) * 100);  
    printf("Your spendings are \$%.2f which is %.2f%% of your income.\n", spendings, spending_percentage);  
}  
  
int main() {  
    
    float income, rent, utilities, groceries, transport, savings, spendings, rent_percentage, utilities_percentage, groceries_percentage, transport_percentage, spending_percentage;  
  
    income = get_user_inputs();  
    

    calculate_percentages(income, rent, utilities, groceries, transport, &savings, &spendings, &rent_percentage, &utilities_percentage, &groceries_percentage, &transport_percentage, &spending_percentage);  
    print_results(savings, rent, rent_percentage, utilities, utilities_percentage, groceries, groceries_percentage, transport, transport_percentage, spendings, spending_percentage);  
  
    return 0;  
}  
